### PR TITLE
Add ability to edit custom token

### PR DESCRIPTION
**Pull request by hewigovens**
Implementation details:

1. Reuse the NewTokenViewController, pass the existed token item as placeholder
2. If user don't change the contract address, it will update the custom token thanks to Realm

### CODE REVIEW

File: Trust/Tokens/Coordinators/TokensCoordinator.swift [hunk 1]
```diff
@@ -59,6 +60,21 @@ class TokensCoordinator: Coordinator {
...
+    func editTokenViewController(token: TokenItem) -> NewTokenViewController {
+        let controller = newTokenViewController()
+        switch token {
+        case .token(let token):
+            guard let address = Address(string: token.contract) else {
+                return controller
+            }
+            let token = ERC20Token(contract: address, name: token.name, symbol: token.symbol, decimals: token.decimals)
+            controller.placeholder = token
+        default:
```
Comment (vikmeup) [inline Trust/Tokens/Coordinators/TokensCoordinator.swift]: Can you make switch statement exhaustive?

File: Trust/Tokens/Types/TokensDataStore.swift [hunk 2]
```diff
@@ -140,6 +140,7 @@ class TokensDataStore {
     func updateDelegate() {
         tokensModel.value = enabledObject
         let tokensViewModel = TokensViewModel(
```
Comment (vikmeup) [inline Trust/Tokens/Types/TokensDataStore.swift]: Extra self.

File: Trust/Tokens/ViewModels/TokensViewModel.swift [hunk 3]
```diff
@@ -13,12 +13,15 @@ struct TokensViewModel {
     let tokens: [TokenObject]
     let tickers: [String: CoinTicker]?
     let nonFungibleTokens: [NonFungibleToken]
+    var config: Config?
```
Comment (vikmeup) [inline Trust/Tokens/ViewModels/TokensViewModel.swift]: Can you make config non optional, and default is Confg()?

File: Trust/Tokens/ViewModels/TokensViewModel.swift [hunk 4]
```diff
@@ -13,15 +13,17 @@ struct TokensViewModel {
     let tokens: [TokenObject]
     let tickers: [String: CoinTicker]?
     let nonFungibleTokens: [NonFungibleToken]
-    var config: Config?
+    var config = Config()
```
Comment (vikmeup) [inline Trust/Tokens/ViewModels/TokensViewModel.swift]: can you make it let? 

in initializer you would have `config: Config = Config()` then

File: Trust/Tokens/ViewControllers/NewTokenViewController.swift [hunk 5]
```diff
@@ -11,7 +11,14 @@ protocol NewTokenViewControllerDelegate: class {
 
 class NewTokenViewController: FormViewController {
 
-    let viewModel = NewTokenViewModel()
+    var viewModel = NewTokenViewModel()
+    var placeholder: ERC20Token? {
+        didSet {
+            if self.placeholder != nil {
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/NewTokenViewController.swift]: can you do `guard`?

File: Trust/Tokens/ViewControllers/TokensViewController.swift [hunk 6]
```diff
@@ -132,13 +133,23 @@ extension TokensViewController: UITableViewDelegate {
...
 
     func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
-        return viewModel.canDelete(for: indexPath.row, section: indexPath.section)
+        return viewModel.canEdit(for: indexPath.row, section: indexPath.section)
     }
 
-    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
-        if editingStyle == UITableViewCellEditingStyle.delete {
-            delegate?.didDelete(token: viewModel.item(for: indexPath.row, section: indexPath.section), in: self)
+    func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/TokensViewController.swift]: This is nice! I didn't know there is `editActionsForRowAt `

**Comment by hewigovens** [comment]
👍

**Pull request merged**